errbot/backends/slack.py

```diff
@@ -334,7 +334,15 @@ def _message_event_handler(self, event):
             text = event['text']
             user = event.get('user', event.get('bot_id'))
 
+        mentioned = []
+
+        for word in text.split(' '):
+            if word.startswith('<') or word.startswith('@') or word.startswith('#'):
+                log.debug('Someone mentioned')
+                mentioned.append(self.build_identifier(word.replace(':', '')))
+
         text = re.sub("<[^>]*>", self.remove_angle_brackets_from_uris, text)
+        text = re.sub('<@[^>]*>:*', '', text)
```

Comment:
I find this undesirable. This removes information about the text message that is available to plugins. No other backends do such a thing, so Slack shouldn't become an exception to the rule.

However, we already have convenience functions to change a user id to a username. What you may do is replace `<@U12345>` with `@username` here. That way plugins would receive the same text that a user actually sees in Slack. In 9 out of 10 cases, I'd argue that is what you want.

This would have the added benefit of "magically" making `BOT_ALT_PREFIXES` work on Slack (when setting `BOT_ALT_PREFIXES = ('@yourbotname',)`.